README.md

```diff
@@ -66,139 +69,264 @@ or
-</p>
+## 📝 Customization
 
-You need to load `ts-fold-indicators-mode`:
-- `use-package`
-   ```elisp
-   (use-package ts-fold-indicators
-   :straight (ts-fold-indicators :type git :host github :repo "emacs-tree-sitter/ts-fold"))
-   ```
+Although ts-fold aims to have good folding out of the box for all supported
+definitions, people will indubitably have their own preferences or desired
+functionality. The following section outlines how to add your own folding
+definitions and folding functions to make ts-fold work for you. If there are any
+improvements you find for existing or new languages, please do raise a PR so
+that others may benefit from better folding in the future!
 
--
-   ```elisp
-   (add-to-list 'load-path "/path/to/lib")
-   (require ts-fold)
-   ```
-   or
+### ⚪ Folding on new nodes
 
-   ```elisp
-   (use-package ts-fold-indicators
-      :load-path "/path/to/lib")
-   ```
+Ts-fold defines all its folding definitions in the the variable
+`ts-fold-range-alist` which is an alist with the key of the alist being the
+mode and the value being another alist of fold definitions.
 
-You can then enable this manually by doing the following
-
-```
-M-x ts-fold-indicators-mode
+```elisp
+;; Example of ts-fold-range-alist's structure
+'((c-mode . c-folding-definitions) ;; <language>-folding-definitions is structured as shown below
+  (css-mode . css-folding-definitions)
+  (go-mode . go-folding-definitions)
+  (scala-mode . scala-folding-definitions)
+  ...)
+
+;; Examle of a folding definition alist
+(setq css-folding-definitions
+    (block . ts-fold-range-seq)
+    (comment . ts-fold-range-c-like-comment))
 ```
 
-- To enable this automatically whenever `tree-sitter-mode` is enabled:
+So you can select whatever node that you want to fold on it.
+
+To find what node you'll want to fold closed, refer to the
+[tree-sitter documentation](https://emacs-tree-sitter.github.io/getting-started/#view-the-syntax-tree)
+about viewing nodes. `tree-sitter-debug` and `tree-sitter-query-builder`
+are both very useful for this.
+
+For the folding functions, ts-fold provides some default
+
+- `ts-fold-range-seq` - Folds from the start of the node to the end of the node
+  leaving a buffer of one character on each side. Usually used for code blocks
+  that have bracketing delimiters.
+
+  ```c++
+  int main() { // <-- start of tree-sitter block node
+      printf("Hello, World\n");
+      return 0;
+  } // <-- end of tree-sitter block node
+
+  // |
+  // | '(block . ts-fold-range-seq)
+  // V
+
+  int main() {...} // Folded node
+  ```
+
+- `ts-fold-range-block-comment` - Folds multi-line comments that are of the form
+  `/*...*/`. Should show a summary if the commentary plugin is turned on.
+
+  ```c++
+  /*
+   * The main function that gets run after program is compiled
+   * Doesn't take any parameters
+   */
+  int main() {
+      printf("Hello, World\n");
+      return 0;
+  }
+
+  // |
+  // | '(comment . ts-fold-range-block-comment)
```

Comment:
In `c++`, this should be `ts-fold-range-c-like-comment`. Are you using this for the example? 😕 

https://github.com/emacs-tree-sitter/ts-fold/blob/0bae1476a47b707c6003becd87f4c0bf6cf509ba/ts-fold-parsers.el#L84

Is that the reason why we get the second asterisk? 😆

Reply:
Oh, I guess it was intentional. Sorry for the noise.